src/display/annotation_storage.js

```diff
@@ -176,8 +176,13 @@ class AnnotationStorage {
     const clone = new Map();
 
     for (const [key, val] of this.#storage) {
-      const serialized =
-        val instanceof AnnotationEditor ? val.serialize() : val;
+      let serialized = val;
+      if (val instanceof AnnotationEditor) {
+        serialized = val.serialize();
+        if (!val.hasElementChanged(serialized)) {
+          serialized = null;
```

Comment:
This can be slightly shorter as:
```suggestion
        if (val.hasElementChanged(serialized)) {
          continue;
```